### PR TITLE
Adding tests for Short Code Resolver

### PR DESCRIPTION
**Pull request by DillonAd**
I believe I covered the cases in #5, but if there is anything else you want to test let me know!

### CODE REVIEW

File: src/HexMaster.ShortLink.Tests/ResolveShortCodeFunction_Tests.cs [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+                     .ReturnsAsync(segmentMock);
+
+            var logger = NullLogger.Instance;
+
+            // Act
+            var result = await ResolveShortCodeFunction.ResolveShortCode(request, collectorMock.Object, tableMock.Object, "CODE", logger);
+
+            // Assert
+            Assert.IsInstanceOf<RedirectResult>(result);
+            var r = result as RedirectResult;
```
Comment (nikneem) [inline src/HexMaster.ShortLink.Tests/ResolveShortCodeFunction_Tests.cs:50]: r is not a convenient name, should be something like redirectResult

File: src/HexMaster.ShortLink.Tests/ResolveShortCodeFunction_Tests.cs [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+            Assert.AreEqual("https://app.4dn.me/", r.Url);
+        }
+
+        [Test]
+        public async Task ResolveShortCode_Fail_CodeNotFound()
+        {
+            // Assemble
+            var targetEndpoint = "https://google.com";
```
Comment (nikneem) [inline src/HexMaster.ShortLink.Tests/ResolveShortCodeFunction_Tests.cs:94]: This var is never used

File: src/HexMaster.ShortLink.Tests/ResolveShortCodeFunction_Tests.cs [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+                     .ReturnsAsync(segmentMock);
+
+            var logger = NullLogger.Instance;
+
+            // Act
+            var result = await ResolveShortCodeFunction.ResolveShortCode(request, collectorMock.Object, tableMock.Object, code, logger);
+
+            // Assert
+            Assert.IsInstanceOf<RedirectResult>(result);
+            var r = result as RedirectResult;
```
Comment (nikneem) [inline src/HexMaster.ShortLink.Tests/ResolveShortCodeFunction_Tests.cs:86]: Again, naming here

File: src/HexMaster.ShortLink.Tests/ResolveShortCodeFunction_Tests.cs [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+                     .ReturnsAsync(segmentMock);
+
+            var logger = NullLogger.Instance;
+
+            // Act
+            var result = await ResolveShortCodeFunction.ResolveShortCode(request, collectorMock.Object, tableMock.Object, "CODE", logger);
+
+            // Assert
+            Assert.IsInstanceOf<RedirectResult>(result);
+            var r = result as RedirectResult;
```
Comment (nikneem) [inline src/HexMaster.ShortLink.Tests/ResolveShortCodeFunction_Tests.cs:117]: Again, naming here

File: src/HexMaster.ShortLink.Tests/ResolveShortCodeFunction_Tests.cs [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
```
Comment (nikneem) [inline src/HexMaster.ShortLink.Tests/ResolveShortCodeFunction_Tests.cs:153]: Again, naming here

File: src/HexMaster.ShortLink.Tests/ResolveShortCodeFunction_Tests.cs [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+        }
+
+        [Test]
+        [TestCase(" ")]
+        [TestCase("")]
+        [TestCase(null)]
+        public async Task ResolveShortCode_Fail_EmptyEndpoint(string endpoint)
+        {
+            // Assemble
+            var targetEndpoint = "https://google.com";
```
Comment (nikneem) [inline src/HexMaster.ShortLink.Tests/ResolveShortCodeFunction_Tests.cs:128]: This var is never used

File: src/HexMaster.ShortLink.Tests/ResolveShortCodeFunction_Tests.cs [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+            var context = new DefaultHttpContext();
+            var request = context.Request;
+            
+            var collectorMock = new Mock<IAsyncCollector<LinkClickedMessage>>();
+
+            var entity = new ShortLinkEntity() { EndpointUrl = endpoint };
```
Comment (nikneem) [inline src/HexMaster.ShortLink.Tests/ResolveShortCodeFunction_Tests.cs:134]: This one is never used

**Review by nikneem — COMMENTED** [review]
I'm going to approve this PR after some minor tweaks, but I don't understand why you went for this level of complexity. You're now not testing the Resolver function, but the CORE, which has it's own unit tests.

But I really appreciate your work, was this top of mind knowledge or a lot of google?

**Comment by DillonAd** [comment]
I fixed the naming and removed the unused variables. I also noticed that I forgot to rename the class so I fixed that as well. 

The overall idea was to not alter the existing code if at all possible since it is already functioning correctly. This does touch a couple elements of the framework, but I couldn't find another good way to get to the returned URL without making changes to the existing code.

I've been writing a lot of tests lately, but I did have to spend some time on Google to look up how to mock the `TableQuerySegment` class properly. :smiley:

**Pull request merged**